src/gpu/intel/jit/ir/tensor.hpp

```diff
@@ -326,46 +333,19 @@ class layout_t {
         return ret;
     }
 
-    // Storage size in bytes.
-    dim_t size() const {
-        if (is_empty()) return 0;
-        dim_t max_off = 0;
-        dim_t max_block_size = 0;
-        for (auto &b : blocks_) {
-            max_off += (b.block - 1) * (dim_t)b.stride;
-            max_block_size
-                    = std::max(max_block_size, b.block * (dim_t)b.stride);
-        }
-        dim_t max_elems = std::max(max_off + 1, max_block_size);
-        return max_elems * type().size() / type().packing();
-    }
-
-    // Offset in bytes following the last accessible element.
-    dim_t max_off_bytes(bool ignore_offset = false) const {
-        if (is_empty()) return 0;
-        dim_t max_off = 0;
-        for (auto &b : blocks_) {
-            max_off += (b.block - 1) * (dim_t)b.stride;
-        }
-        dim_t after_last = max_off + 1;
-        if (!ignore_offset) after_last += expr_cast<dim_t>(offset_);
-        return after_last * type().size() / type().packing();
-    }
-
     template <typename T = expr_t>
     T offset(const coord_t &args = {}, bool ignore_offset = false) const {
         if (args.is_empty()) return expr_cast<T>(offset_);
 
         expr_t off = 0;
         auto _args = args;
-        for (auto &eb : enumerated_blocks()) {
-            auto &b = eb.second;
+        for (auto &b : blocks()) {
             if (!_args.has(b.dim)) continue;
             auto &idx = _args[b.dim];
             if (is_zero(idx)) continue;
 
             // Do not use modulus for outermost blocks.
-            auto i = is_outermost(eb) ? idx : (idx % b.block);
+            auto i = is_outermost(b) ? idx : (idx % b.block);
```

Comment:
Should we rename `layout_block_t::block` to `layout_block_t::size`?
This is unrelated to the PR, just wanted to bring this up since you are working on stabilizing the API.

Reply:
I agreed and renaming `layout_block_t::{block -> size}` and additionally `layout_block_t::{dim->idx}` are both changes I would like to see as well.